wandb/sdk/launch/utils.py

```diff
@@ -149,7 +149,7 @@ def construct_launch_spec(
         launch_spec["docker"]["docker_image"] = docker_image
 
     if "resource" not in launch_spec:
-        launch_spec["resource"] = resource or "local"
+        launch_spec["resource"] = resource
```

Comment:
If there's no DRC, do we handle it gracefully?

Reply:
unfortunately this is going to have to happen on the backend with a hard error not a warning, unless we want to make a query for the existence of a DRC in the SDK before pushing to a queue.

Reply:
I was planning on actually fixing this in this pr: https://github.com/wandb/wandb/pull/4516, maybe I should just combine them? Or at least merge them at the same time.

Reply:
Actually the "push_to_queue" helper function in `launch_add` handles this I believe. and in `launch.py` we default to "local-container" so it should never be None.